### ISSUE TITLE
Prune external dependencies with unused imports

### ISSUE THREAD

**Issue by Rich-Harris** [rollup/rollup#1352]
[REPL](https://rollupjs.org/repl?version=0.41.6&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QiUyMHVudXNlZCUyMCU3RCUyMGZyb20lMjAnZXh0ZXJuYWwnJTNCJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmNqcyUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJtb2R1bGVJZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE) — if you import from external modules but don't use the imports, it would be nice to be able to remove the resulting `require` statements from the generated bundle. This would have to be configured, since by default Rollup has to assume that the external modules have side-effects.

**Comment by rtsao** [comment]
I'm interested in helping make this happen. I'd be happy to start working on a PR.

**How should this option be configured?**

I think something like this would be reasonable:
```js
{
  // The default option. Assume all external modules have side effects and don't remove unused imports
  pureExternalImports: false, 

  // Assume all external modules don't have side effects and remove unused imports
  pureExternalImports: true, 

  // Assume all external modules but 'foo' and 'bar' have imports with side effects
  pureExternalImports: {
     pure: ['foo', 'bar'] 
  },

  // Assume all external modules but 'foo' and 'bar' have pure imports
  pureExternalImports: {
     impure: ['foo', 'bar'] 
  },

  // Note: if both 'pure' and 'impure' options are provided, an error is thrown
}
```

**Comment by Rich-Harris** [comment]
Fixed in 0.43 — thanks

**Comment by rtsao** [comment]
Looks like webpack is supporting a `"pure-module"` field in the package.json: https://github.com/webpack/webpack/tree/feature/pure-module/examples/pure-module
